### ISSUE TITLE
Invite users to a shared collection

### ISSUE THREAD

**Issue by bumi** [bumi/tasveer#25]
In order to create a collaborative collection I want to invite multiple users to a collection

As we don't want to require any signup/login the idea was to send an email with a secret deep-link. Once the invited user clicks on the link the application opens and does a request to the server (invitations#accept).

If the user does not have the app installed we first ask the user to install the app. 

Alternatively to send the deep-link directly via email we can have a webpage with the deep link and the link in the email redirects the user to the that website. (I guess this might be preferred?)

**Comment by ludoded** [comment]
@bumi 
Alice invites Bob to share the same collection

If Bob doesn’t have the app installed on the device then the URL in the email won’t work.
If we care about such case, then the link in the mail should redirect to a page which will check if the app is installed on the device. If it’s installed then open the app with parameters, if not then redirect to the App Store, so user can install the app and then open again from the link in the mail, to confirm the invitation.

https://stackoverflow.com/a/13196998/1453835 -> this is one way how to implement it on a web page.


To setup the invite scheme url we’d need to put the token parameter and collection_id
I.e. tasveer://invite/details?token={token}&collection_id={id}

Token can indicates the uniqueness of the invite and also we can refer the expiration date to the token. 

When Bob press to that link in the mail, it will open the app and I will get the token and collection_id parameter. I will send those params in the request to confirm that token is valid and invite is correct and if everything is valid, then I will get 200 status and show Bob that invite was accepted.

**Comment by bumi** [comment]
yep, we should add a page on which we should all required information. 
that's a good hack there described on stackoverflow. 

I will setup the page and redirect the user to an URL as defined by you. 
shall we use your suggested example above?

**Comment by ludoded** [comment]
Yes, let’s use the suggested example

**Comment by bumi** [comment]
heja @ludoded when do you have time for this?

**Comment by ludoded** [comment]
Hey @bumi, I have time now, on it

**Comment by ludoded** [comment]
@bumi seems email delivery to an invitee is not implemented yet?

**Comment by bumi** [comment]
The text and template is not done yet. but delivery is on. For testing I need to add your email to the recipient list. can you send me the email you want to use?

**Comment by bumi** [comment]
Also I saw this: https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html - is that interesting for us?

**Comment by ludoded** [comment]
@bumi dos231@gmail.com and selivanovhelen@gmail.com

**Comment by ludoded** [comment]
> Also I saw this: https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html - is that interesting for us?

I'll take a look shortly and let you know
